assets/js/atomic/blocks/product-elements/image/block.tsx

```diff
@@ -17,39 +16,74 @@ import {
 } from '@woocommerce/base-hooks';
 import { withProductDataContext } from '@woocommerce/shared-hocs';
 import { useStoreEvents } from '@woocommerce/base-context/hooks';
+import type { HTMLAttributes } from 'react';
 
 /**
  * Internal dependencies
  */
 import ProductSaleBadge from '../sale-badge/block';
 import './style.scss';
+import type { BlockAttributes } from './types';
 
-/**
- * Product Image Block Component.
- *
- * @param {Object}            props                           Incoming props.
- * @param {string}            [props.className]               CSS Class name for the component.
- * @param {string|undefined}  [props.imageSizing]             Size of image to use.
- * @param {boolean|undefined} [props.showProductLink]         Whether or not to display a link to the product page.
- * @param {boolean}           [props.showSaleBadge]           Whether or not to display the on sale badge.
- * @param {string|undefined}  [props.saleBadgeAlign]          How should the sale badge be aligned if displayed.
- * @param {boolean}           [props.isDescendentOfQueryLoop] Whether or not be a children of Query Loop Block.
- * @return {*} The component.
- */
-export const Block = ( props ) => {
+const ImagePlaceholder = (): JSX.Element => {
+	return (
+		<img src={ PLACEHOLDER_IMG_SRC } alt="" width={ 500 } height={ 500 } />
+	);
+};
+
+type ImageProps = {
+	image: {
+		alt?: string;
+		id: number;
+		name: string;
+		sizes: string;
+		src: string;
+		srcset: string;
+		thumbnail: string;
+	};
+	loaded: boolean;
+	showFullSize: boolean;
+	fallbackAlt: string;
+};
+
+const Image = ( {
+	image,
+	loaded,
+	showFullSize,
+	fallbackAlt,
+}: ImageProps ): JSX.Element => {
+	const { thumbnail, src, srcset, sizes, alt } = image || {};
+	const imageProps = {
+		alt: alt || fallbackAlt,
+		hidden: ! loaded,
+		src: thumbnail,
+		...( showFullSize && { src, srcSet: srcset, sizes } ),
+	};
+
+	return (
+		<>
+			{ imageProps.src && (
```

Comment:
It seems that in the type above `src` is required. So, if types were respected, this would always be true, or we'd have a bunch of `//ts-ignore` where this is not guaranteed.

So we should account for this being `undefined` in the type definition as well.

Reply:
I'm not sure what you you mean in this case. Can you give me a bit more context, @sunyatasattva?

Reply:
I mean, if the component is checking here that `imageProps.src` exists, I assume that there are circumstances in which it is `undefined`. So the type definition above should mark it as such. Does it make sense?

Reply:
Yes, now I know what you mean, and I adjusted the corresponding attributes accordingly.

Reply:
Hey @nielslange, out of curiosity, where did you adjust these attributes? Sorry for being annoying, I just know that this type is going to be a problem for me soon (when we implement image placeholders on the PQ block) if it's not adjusted.